### PR TITLE
remove moved directories from default config

### PR DESCRIPTION
**Pull request by jadwigo**
the mentioned directories are moved outside the webroot - so they only interfere with legitimate site content.

### CODE REVIEW

**Comment by GawainLynch** [comment]
Not really sure completely removing it is the right approach … but close to. Ambivalent about this one though.

Technically they are out of the web root in nearly all cases, the general exception being people stuck on crappy hosting … and that's all Apache.

@CarsonF & @bobdenotter

**Comment by CarsonF** [comment]
My uninformed view is anyone using nginx knows what they are doing..therefore doesn't have their web root the same as their project root.

**Comment by jadwigo** [comment]
Also - a few lines earlier executing php files in the apps, extensions and other directories is excluded anyway by
```
    # Block PHP files from being run in upload (files), app, theme and extension directories
    location ~* /(?:app|extensions|files|theme)/(.*)\.php$ {
        deny all;
    }
```

**Comment by SahAssar** [comment]
I think it should stay. Making an assumption that just because someone is use certain software they will be competent is dangerous and if someone uses a git install (I know that is discouraged for non-dev work, but still) the folders will be there.

**Comment by CarsonF** [comment]
Here's a novel idea: We could change our git repo to not have the web root be the same as the project root. smh

**Comment by SahAssar** [comment]
Sure, but until that change is in perhaps we should keep this in?

**Comment by jadwigo** [comment]
If we keep these lines we also need to tell users that any slug may not contain app/ src/ tests/ vendor/ ... because those will give you a HTTP 403 in nginx

In a normal install these lines will actually decrease the usability of bolt. It will for example prevent content types with the slug apps .. because the detail page http://example.com/app/pokemon-go will be forbidden and it will also block any javascript in a vendor or src folder like some frameworks like to do.


```
    # Block access to the app, cache & vendor directories		
    location ~ /(?:app|src|tests|vendor)/(.*)$ {		
        deny all;		
    }
```

And because it is a webserver 403 it is nigh impossible to debug.

The actual directory structure has changed, removing the corresponding lines places the burden on the people who move their webroot to something not standard instead of the users who don't move stuff around.

Keeping these lines in will make it harder for people who install bolt the recommended way on nginx. 

Also this is just copy pasted from the bolt 2 config - and keeping copypasta around because it's already there is not good.😕

**Comment by GawainLynch** [comment]
> The actual directory structure has changed, removing the corresponding lines places the burden on the people who move their webroot to something not standard instead of the users who don't move stuff around.

Yes, and no. Un-commented, I'll give you that as a point. But the plan all along has been to update these so it is a bit more informative … especially when it comes to securing your site. 

> Also this is just copy pasted from the bolt 2 config - and keeping copypasta around because it's already there is not good.

Yes … and I am to blame, as I have had a TODO forever … but my hand was forced on a release date and I haven't stopped since … but if I was to take the suggested approach with core code, I'm pretty sure I'd lose the respect of our fellow team members — and commit access — pretty quickly. :wink:

**Comment by GawainLynch** [comment]
OK, how about this for a compromise?
```
    # Block access to the app, cache & vendor directories
    #
    # NOTE: If you do *not* have your web root as a sub-directory of your 
    # installation root, you should uncomment this location block to prevent
    # site visitors having access to the various directories that contain
    # executable code.
    #
    # NOTE: This approach is *not* recommended for production use.
    #
    #location ~ /(?:app|src|tests|vendor)/(.*)$ {
    #    deny all;
    #}
```

**Comment by SahAssar** [comment]
WFM 👍

**Comment by jadwigo** [comment]
I would rephrase that a little ... 
```
    # Block access to the app, cache & vendor directories
    #
    # NOTE: If you have one or more of 'app' , 'src', 'test' and 'vendor' as 
    # sub-directories of your installation root, you should uncomment
    # this location block to prevent site visitors having access to the 
    # various directories that contain executable code.
    #
    # NOTE: This approach is *not* recommended for production use.
    #
    #location ~ /(?:app|src|tests|vendor)/(.*)$ {
    #    deny all;
    #}
```

**Comment by GawainLynch** [comment]
> I would rephrase that a little ...

I like *much* more than mine! 

Are you good to update it, or would you like me to ninja it in on the branch rebase?

**Comment by jadwigo** [comment]
ninja away :)

**Comment by GawainLynch** [comment]
Merged into `release/3.0` via 3201d649b84a021aad30177950bc41f907562413
